controllers/space/space_controller_test.go

```diff
@@ -42,228 +42,259 @@ func TestCreateSpace(t *testing.T) {
-			HasStateLabel("cluster-assigned").
-			HasFinalizer()
-		nsTmplSet := nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, "oddity", member1.Client).
-			Exists().
-			HasTierName(base1nsTier.Name).
-			HasClusterResourcesTemplateRef("base1ns-clusterresources-123456new").
-			HasNamespaceTemplateRefs("base1ns-code-123456new", "base1ns-dev-123456new", "base1ns-stage-123456new").
-			Get()
-		AssertThatCountersAndMetrics(t).
-			HaveSpacesForCluster("member-1", 1).
-			HaveSpacesForCluster("member-2", 0) // check that increment for member-1 was called
-
-		t.Run("requeue while NSTemplateSet is not ready", func(t *testing.T) {
-			// given another round of requeue without while NSTemplateSet is *not ready*
-			nsTmplSet.Status.Conditions = []toolchainv1alpha1.Condition{
-				nstemplatetsettest.Provisioning(),
-			}
-			err := member1.Client.Update(context.TODO(), nsTmplSet)
-			require.NoError(t, err)
-			ctrl := newReconciler(hostClient, member1, member2)
-
-			// when
-			res, err := ctrl.Reconcile(context.TODO(), requestFor(s))
-
-			// then
-			require.NoError(t, err)
-			assert.False(t, res.Requeue)
-			spacetest.AssertThatSpace(t, test.HostOperatorNs, "oddity", hostClient).
-				Exists().
-				HasStatusTargetCluster("member-1").
-				HasConditions(spacetest.Provisioning()).
-				HasStateLabel("cluster-assigned").
-				HasFinalizer()
-			nsTmplSet := nstemplatetsettest.AssertThatNSTemplateSet(t, test.MemberOperatorNs, "oddity", member1.Client).
-				Exists().
-				HasClusterResourcesTemplateRef("base1ns-clusterresources-123456new").
-				HasNamespaceTemplateRefs("base1ns-code-123456new", "base1ns-dev-123456new", "base1ns-stage-123456new").
-				Get()
-			AssertThatCountersAndMetrics(t).
-				HaveSpacesForCluster("member-1", 1).
-				HaveSpacesForCluster("member-2", 0) // nothing has changed since previous increment
+		type testRun struct {
+			name           string
+			featureToggles string
+		}
 
-			t.Run("when NSTemplateSet is ready update the provisioned namespace list", func(t *testing.T) {
-				// given another round of requeue without with NSTemplateSet now *ready*
-				nsTmplSet.Status.Conditions = []toolchainv1alpha1.Condition{
-					nstemplatetsettest.Provisioned(),
-				}
-				nsTmplSet.Status.ProvisionedNamespaces = []toolchainv1alpha1.SpaceNamespace{
-					{
-						Name: "john-dev",
-						Type: "default",
-					},
-					{
-						Name: "john-stage",
-					},
+		tests := []testRun{
+			{
+				name: "without feature toggles",
+			},
+			{
+				name:           "with feature toggles",
+				featureToggles: "feature-1,feature-2",
+			},
+		}
+		for _, testRun := range tests {
+			t.Run(testRun.name, func(t *testing.T) {
+				spaceOptions := []spacetest.Option{spacetest.WithSpecTargetCluster("member-1")}
+				if testRun.featureToggles != "" {
+					spaceOptions = append(spaceOptions, spacetest.WithAnnotation(toolchainv1alpha1.FeatureToggleNameAnnotationKey, testRun.featureToggles))
 				}
```

Comment:
This is the first change in this test: Adding the feature annotation to the Space depending on the specific test from the table.